lib/AST/ExternalASTMerger.cpp

```diff
@@ -144,14 +144,14 @@ class LazyASTImporter : public ASTImporter {
     }
     if (auto *ToTag = dyn_cast<TagDecl>(To)) {
       ToTag->setHasExternalLexicalStorage();
-      ToTag->setMustBuildLookupTable();
+      ToTag->getPrimaryContext()->setMustBuildLookupTable();
       assert(Parent.CanComplete(ToTag));
     } else if (auto *ToNamespace = dyn_cast<NamespaceDecl>(To)) {
       ToNamespace->setHasExternalVisibleStorage();
       assert(Parent.CanComplete(ToNamespace));
     } else if (auto *ToContainer = dyn_cast<ObjCContainerDecl>(To)) {
       ToContainer->setHasExternalLexicalStorage();
-      ToContainer->setMustBuildLookupTable();
+      ToContainer->getPrimaryContext()->setMustBuildLookupTable();
```

Comment:
How are these changes related?

Reply:
By removing the `ImportDefinitionIfNeeded` from `GetAlreadyImportedOrNull` a new assertion is triggered here. `setMustBuildLookupTable` must alwasy be called on a primary context.